### DISCUSSION TITLE
Show nutrition info when available, and always show in edit mode

### DISCUSSION THREAD

**Discussion by epheterson** [mealie-recipes/mealie#1401] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [ X ] Reviewed the documentation and verified that my feature request isn't already a feature.
- [ X ] Reviewed the existing feature requests and my feature has not been requested.
- [ X ] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Many recipes include detailed nutrition info and/or calories per serving. There's currently no way to store this in Mealie outside of the Notes section, and the data cannot be automatically scraped.

**(Optional): Suggest A Solution**
It'd be great if there were a field specifically for nutrition info so that we could copy/paste data into it and have it separately displayed, without having to custom-title a Notes field.

Further, it'd be great if this information could be extracted automatically from recipes that include it. The extraction could be tricky since sites display nutrition info differently. They may include only calories, text-based nutrition info (string) text-based nutrition info (table), or nutrition label style as an image. Support for all styles would be ideal.

**Additional Information**
_FILL THESE OUT_
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by hay-kot** [comment]
This already exists and is implemented. You need to enable Nutrition details in the settings of the recipe for them to show.

**Comment by epheterson** [comment]
Apologies! I looked around for it and didn't find it, though now I see it and a brief mention in the docs.

Is there any way to have this always show in edit view, and to conditionally show in all recipes when the data exists?

It appears that so far the info is only visible in edit view, and only after enabling that toggle, which really limits its usefulness (unless I'm missing something).

**Comment by epheterson** [comment]
Should we use this to track the bug that nutrition info doesn't show, even while enabled (outside of edit mode)?

**Reply by hay-kot** [reply]
Best to create an issue, I don't check these as often as I check issues, especially when I'm working on fixes. Case and point is that I've already forgotten about this issue twice 😄

**Reply by epheterson** [reply]
Filed it here! https://github.com/hay-kot/mealie/issues/1438

**Comment by epheterson** [comment]
This now shows up but includes all nutrition properties even if only a subset is filled out. It’d be nice if it showed only nutrition data that has a value and hides the rest.

**Comment by michael-genson** [comment]
Not sure when, but this was eventually implemented; only nutrition options with values are shown:
![image](https://github.com/user-attachments/assets/4a6ff8fe-09ba-4473-aabb-55d4816f18bf)
